master/internal/db/postgres_trial_intg_test.go

```diff
@@ -917,7 +910,52 @@ func TestGenericMetricsIO(t *testing.T) {
 	require.EqualValues(t, trialRunID, metricReports[0].TrialRunId)
 	require.EqualValues(t, batches, metricReports[0].TotalBatches)
 	require.EqualValues(t, tr.ID, metricReports[0].TrialId)
-	require.Equal(t, metrics, metricReports[0].Metrics.Fields["avg_metrics"].GetStructValue())
+	require.Equal(t, metrics, metricReports[0].Metrics.
+		Fields[model.TrialMetricsJSONPath(false)].GetStructValue())
+
+	// test generic metrics summary metric write and read.
+	metrics2, err := structpb.NewStruct(map[string]any{"aloss": 20, "bloss": 30})
+	require.NoError(t, err)
+	trialMetrics2 := trialMetrics
+	trialMetrics2.StepsCompleted = int32(batches * 2)
+	trialMetrics2.Metrics = &commonv1.Metrics{AvgMetrics: metrics2}
+	err = db.AddTrialMetrics(ctx, trialMetrics, "inference")
+	require.NoError(t, err)
+
+	query := fmt.Sprintf(`SELECT name,
+summary_metrics->'%[1]s'->name->>'max' AS max,
+summary_metrics->'%[1]s'->name->>'min' AS min,
+summary_metrics->'%[1]s'->name->>'sum' AS sum,
+summary_metrics->'%[1]s'->name->>'last' AS last,
+summary_metrics->'%[1]s'->name->>'count' AS count,
+summary_metrics->'%[1]s'->name->>'type' AS type
+FROM trials
+CROSS JOIN jsonb_object_keys(summary_metrics->'%[1]s') AS name
+WHERE id = ?
+ORDER BY name ASC`, model.TrialSummaryMetricsJSONPath("inference"))
```

Comment:
maybe I'd even just do this here instead:
```suggestion
ORDER BY name ASC`, "inference")
```

to ensure that the database structure for these is explicitly what we want.

Reply:
good idea